posts/2023-12-08-cargo-cache-cleaning.md

```diff
@@ -0,0 +1,182 @@
+---
+layout: post
+title: "Cargo cache cleaning"
+author: Eric Huss
+team: The Cargo Team <https://www.rust-lang.org/governance/teams/dev-tools#cargo>
+---
+
+Cargo has recently gained an unstable feature on the nightly channel (starting with nightly-2023-11-17) to perform automatic cleaning of cache content within Cargo's home directory.
+This post includes:
+
+- A description of what this means ([What is this feature?](#what-is-this-feature))
+- A call for help ([What to watch out for](#what-to-watch-out-for) and [Request for feedback](#request-for-feedback))
+- Implementation details ([Design considerations and implementation details](#design-considerations-and-implementation-details))
+- An outline of what is planned for the future which includes cleaning of target directories ([Plan for the future](#plan-for-the-future))
+
+In short, we are asking people who use the nightly channel to enable this feature and report any issues you encounter on the [Cargo issue tracker][cargo-issues].
+To enable it, place the following in your Cargo config file (typically located in `~/.cargo/config.toml` or `%USERPROFILE%\.cargo\config.toml` for Windows):
+
+```toml
+[unstable]
+gc = true
+```
+
+Or set the `CARGO_UNSTABLE_GC=true` environment variable or use the `-Zgc` CLI flag to turn it on for individual commands.
+
+We'd particularly like people who use unusual filesystems or environments to give it a try, since there are some parts of the implementation which are sensitive and need battle testing before we turn it on for everyone.
+
+[cargo-issues]: https://github.com/rust-lang/cargo/issues/
+
+## What is this feature?
+
+Cargo keeps a variety of cached data within the Cargo home directory.
+This cache can grow unbounded and can get quite large (easily reaching many gigabytes).
+Community members have developed tools to manage this cache, such as [`cargo-cache`](https://crates.io/crates/cargo-cache), but cargo itself never exposed any ability to manage it.
+
+This cache includes:
+
+- [Registry index data], such as package dependency metadata from [crates.io].
+- Compressed `.crate` files downloaded from a registry.
+- The uncompressed contents of those `.crate` files, which `rustc` uses to read the source and compile dependencies.
+- Clones of git repositories used by git dependencies.
+
+The new garbage collection ("GC") feature adds tracking of this cache data so that cargo can automatically or manually remove unused files.
+It keeps an SQLite database which tracks the last time the various cache elements have been used.
+Every time you run a cargo command that reads or writes any of this cache data, it will update the database with a timestamp of when that data was last used.
+
+[crates.io]: https://crates.io/
```

Comment:
```suggestion
Every time you run a cargo command that reads or writes any of this cache data, it will update the database with a timestamp of when that data was last used.

What isn't yet included is cleaning of target directories, see [Plan for the future](#plan-for-the-future).

[crates.io]: https://crates.io/
```
Had better wording for this the other day but close enough.

I feel like just after a list of what caches are being tracked/gced  is a good time to bring up what we aren't GCing as that is when the reader will notice and ask questions